CPAC/nuisance/utils/utils.py

```diff
@@ -0,0 +1,870 @@
+                                ]
+                                + (lat_ven_mni_to_anat, "reference_image")
+                            )
+                        )
+
+                    pipeline_resource_pool[ventricles_key] = (
+                        lat_ven_mni_to_anat,
+                        "output_image",
+                    )
+
+                else:
+                    # perform the transform using FLIRT
+                    lat_ven_mni_to_anat = pe.Node(
+                        interface=fsl.ApplyWarp(),
+                        name="{}_fsl_applywarp".format(ventricles_key),
+                    )
+                    lat_ven_mni_to_anat.inputs.interp = "nn"
+
+                    nuisance_wf.connect(
+                        *(
+                            transforms["mni_to_anat_linear_xfm"]
+                            + (lat_ven_mni_to_anat, "field_file")
+                        )
+                    )
+                    nuisance_wf.connect(
+                        *(
+                            pipeline_resource_pool["Ventricles"]
+                            + (lat_ven_mni_to_anat, "in_file")
+                        )
+                    )
+                    nuisance_wf.connect(
+                        *(
+                            pipeline_resource_pool[mask_key]
+                            + (lat_ven_mni_to_anat, "ref_file")
+                        )
+                    )
+
+                    pipeline_resource_pool[ventricles_key] = (
+                        lat_ven_mni_to_anat,
+                        "out_file",
+                    )
+
+            if csf_mask_exist:
+                # reduce CSF mask to the lateral ventricles
+                mask_csf_with_lat_ven = pe.Node(
+                    interface=afni.Calc(outputtype="NIFTI_GZ"),
+                    name="{}_Ventricles".format(mask_key),
+                )
+                mask_csf_with_lat_ven.inputs.expr = "a*b"
+                mask_csf_with_lat_ven.inputs.out_file = "csf_lat_ven_mask.nii.gz"
+
+                nuisance_wf.connect(
+                    *(
+                        pipeline_resource_pool[ventricles_key]
+                        + (mask_csf_with_lat_ven, "in_file_a")
+                    )
+                )
+                nuisance_wf.connect(
+                    *(
+                        pipeline_resource_pool[mask_key]
+                        + (mask_csf_with_lat_ven, "in_file_b")
+                    )
+                )
+
+                pipeline_resource_pool[
+                    "{}_Unmasked".format(mask_key)
+                ] = pipeline_resource_pool[mask_key]
+                pipeline_resource_pool[mask_key] = (mask_csf_with_lat_ven, "out_file")
+
+            else:
+                pipeline_resource_pool[mask_key] = pipeline_resource_pool[
+                    ventricles_key
+                ]
+
+        return pipeline_resource_pool
+    return None
```

Comment:
It hasn't seemed to cause a problem, but I don't understand why this case would return `None` instead of `pipeline_resource_pool` or `{}`

Reply:
If no one knows you could turn it into an error? Maybe it just never happens?

Reply:
#2079